go-controller/pkg/node/iptables/iptables_manager.go

```diff
@@ -0,0 +1,402 @@
+package iptables
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/util/iptables"
+	kexec "k8s.io/utils/exec"
+)
+
+type Chain struct {
+	Table iptables.Table
+	Chain iptables.Chain
+	Proto iptables.Protocol
+}
+
+func (c Chain) equal(c2 Chain) bool {
+	if c.Table != c2.Table {
+		return false
+	}
+	if c.Chain != c2.Chain {
+		return false
+	}
+	if c.Proto != c2.Proto {
+		return false
+	}
+	return true
+}
+
+// RuleArgs represents an iptables entry
+type RuleArgs struct {
+	Args []string
+}
+
+func (r RuleArgs) equal(r2 RuleArgs) bool {
+	var found bool
+	if len(r.Args) != len(r2.Args) {
+		return false
+	}
+	// ensure all args for r are found in rule2 args
+	for _, ruleArg := range r.Args {
+		found = false
+		for _, rule2Arg := range r2.Args {
+			if ruleArg == rule2Arg {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	// ensure all args for rule2 are found in r args
```

Comment:
Why is the second loop required? if `len(r.Args) == len(r2.Args)` and `r2.Args` contains all of the elements of `r.Args` then the slices are equal?

Also, can we just use [reflect.DeepEqual](http://golang.org/pkg/reflect/#DeepEqual) on the slices or the whole objects?

Reply:
replaced with `DeepEqual`.